data_steward/test/unit_test/vocabulary_test.py

```diff
@@ -0,0 +1,97 @@
+import unittest
+import StringIO
+from vocabulary import *
+from vocabulary import _transform_csv
+import test_util
+import os
+import mock
+
+
+class VocabularyTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        super(VocabularyTest, self).setUp()
+
+    @staticmethod
+    def do_transform_file(input_text):
+        in_fp = StringIO.StringIO(input_text)
+        out_fp = StringIO.StringIO()
+        _transform_csv(in_fp, out_fp)
+        return out_fp.getvalue()
+
+    def test_transform_file(self):
+        header = ['concept_id', 'valid_start_date', 'valid_end_date', 'invalid_reason']
+        row1 = ['1', '2017-05-17', '2099-12-31', '']
+        row2 = ['2', '2019-01-01', '2099-12-31', '']
+        rows = [header, row1, row2]
+        lines = map(lambda r: DELIMITER.join(r), rows)
```

Comment:
Codacy is indicating this lamba expression could be replaced by:
`lines = [DELIMITER.join(row) for row in rows]`

For this small example, it isn't much of a performance hit, but generally, the list comprehension is faster than the map function.  You can test it yourself with a python interpreter.

`import timeit`
`timeit.timeit('map(lambda r: "\t".join(r), rows)', setup="rows=[['concept_id', 'valid_start_date', 'valid_end_date', 'invalid_reason'], ['1', '2017-05-17', '2099-12-31', ''], ['2', '2019-01-01', '2099-12-31', '']]", number=5000)`
`timeit.timeit('["\t".join(row) for row in rows]', setup="rows=[['concept_id', 'valid_start_date', 'valid_end_date', 'invalid_reason'], ['1', '2017-05-17', '2099-12-31', ''], ['2', '2019-01-01', '2099-12-31', '']]", number=5000)`

As mentioned, for this small example, there isn't a noticable change, but it is there.